src/modules/sni/item.cpp

```diff
@@ -409,8 +411,44 @@ void Item::makeMenu() {
 }
 
 bool Item::handleClick(GdkEventButton* const& ev) {
+  int x = ev->x_root;
+  int y = ev->y_root;
+  auto window = gdk_window_get_toplevel(ev->window);
```

Comment:
Most of the values here don't change - each bar instance is pinned to an output and the screen edge. The only parts that can change are window width/height. You can calculate the values on `Bar::window::signal_configure_event` and cache as `Bar::x_global/y_global`.

And the calculations in the event handler could be reduced to `(bar.x_global + ev.x_root, bar.y_global + ev.y_root)`.

Reply:
`x_root` and `y_root` is the position of the cursor in the click event, that is supposed to change between every event. I think that monitor geometry can change between events as well.

Reply:
Monitor geometry can change, but it will cause the bar to adjust size and resend the configure event. So it's still safe to cache the global position of a window and only add x_root/y_root to that in the event handler.

Reply:
Yeah, but `signal_configure_event` ain't enough, since if one specifies both width and height, then the `configure` event won't get called.